### PR TITLE
HW06 partial answers

### PR DESCRIPTION
**Pull request by hatsy**

### CODE REVIEW

File: HW06_ex09_02.py [hunk 1]
```diff
@@ -13,11 +13,26 @@
...
+	if word.find("e") == -1:
+		return True
+	else:
+		return False
 
 
 ##############################################################################
 def main():
-    pass  # Call your function(s) here.
+	with open("words.txt", "r") as fin:
```
Comment (danielsgriffin) [inline HW06_ex09_02.py]: Good.

File: HW06_ex09_03.py [hunk 2]
```diff
@@ -17,13 +17,69 @@
...
 # Body
+def avoids(word, forbidden_letters):
+	""" Returns True if the word does not contain the forbidden letters.
+
+	Takes a word and a string of forbidden letters, and that returns True 
+	if the word doesn't use any of the forbidden letters."""
+
+	if len(forbidden_letters) <= 0:
+		return True
+	elif (word.find(forbidden_letters[0]) == -1) and (avoids(word, forbidden_letters[1:])): #Use recursion
```
Comment (danielsgriffin) [inline HW06_ex09_03.py]: interesting. [I have no expertise here] I wonder how optimized this is v. looping through and using `in` or looping through and using `.find`

Comment (danielsgriffin) [inline HW06_ex09_03.py]: Keshav also used find.

Comment (danielsgriffin) [inline HW06_ex09_03.py]: http://stackoverflow.com/questions/18437798/find-vs-in-operation-in-string-python

File: HW06_ex09_03.py [hunk 3]
```diff
@@ -17,13 +17,69 @@
...
+		return True
+	else:
+		return False
+
+def words_avoid(words, forbidden_letters):
+	""" Returns the words that avoid the forbidden_letters"""
+	words_avoid = []
+	for word in words:
+		if avoids(word, forbidden_letters) == True:
+			words_avoid.append("word")
```
Comment (danielsgriffin) [inline HW06_ex09_03.py]: is this supposed to be a string?

File: HW06_ex09_03.py [hunk 4]
```diff
@@ -17,13 +17,69 @@
...
+def sort_max_indexes(items_list):
+	indexes = range(len(items_list))
+	indexes.sort(key=items_list.__getitem__)
+	return indexes[-1]
+
+def find_combination(words, forbidden_letters=""):
+	""" Searches for the best combination of letters that excludes the least words.
+
+	This function uses recursion. To avoid computation timeout it excludes all words that contain
+	the forbidden letter when searching for the next letter."""
```
Comment (danielsgriffin) [inline HW06_ex09_03.py]: Interesting.

File: HW06_ex09_03.py (hunk 3 above)
Comment (hrenold) [inline HW06_ex09_03.py]: No it's supposed to be word. Thanks.

**Pull request closed without merging**